client/packages/system/src/Patient/PatientView/AddButton.tsx

```diff
@@ -5,77 +5,98 @@ import {
   SplitButtonOption,
   PlusCircleIcon,
   useUrlQuery,
+  UserStoreNodeFragment,
 } from '@openmsupply-client/common';
 import {
   PatientModal,
   usePatientModalStore,
+  usePatientStore,
 } from '@openmsupply-client/programs';
 import { PatientTabValue } from './PatientView';
+import { useInsuranceProviders } from '../apiModern';
 
 interface AddButtonProps {
   /** Disable the whole control */
   disabled: boolean;
   disableEncounterButton: boolean;
+  store?: UserStoreNodeFragment;
 }
 
 export const AddButton: React.FC<AddButtonProps> = ({
   disabled,
   disableEncounterButton,
+  store,
 }) => {
   const t = useTranslation();
   const { urlQuery, updateQuery } = useUrlQuery();
   const currentUrlTab = urlQuery['tab'];
+  const { createNewPatient } = usePatientStore();
   const { setModal: selectModal, reset } = usePatientModalStore();
+  const {
+    query: { data: InsuranceProvidersData },
+  } = useInsuranceProviders();
 
-  const options: [
-    SplitButtonOption<PatientModal>,
-    SplitButtonOption<PatientModal>,
-    SplitButtonOption<PatientModal>,
-    SplitButtonOption<PatientModal>,
-  ] = useMemo(
-    () => [
-      {
-        value: PatientModal.ProgramSearch,
-        label: t('button.add-program'),
-        isDisabled: false,
-      },
-      {
-        value: PatientModal.Encounter,
-        label: t('button.add-encounter'),
-        isDisabled: disableEncounterButton,
-      },
-      {
-        value: PatientModal.ContactTraceSearch,
-        label: t('button.add-contact-trace'),
-      },
-      { value: PatientModal.Insurance, label: t('button.add-insurance') },
-    ],
-    [disableEncounterButton, t]
-  );
-  const [programOption, encounterOption, contactTraceOption, insuranceOption] =
-    options;
+  const options: SplitButtonOption<PatientModal>[] = useMemo(() => {
+    const baseOptions: SplitButtonOption<PatientModal>[] = [];
+
+    if (store?.preferences.omProgramModule) {
+      baseOptions.push(
+        {
+          value: PatientModal.ProgramSearch,
+          label: t('button.add-program'),
+          isDisabled: false,
+        },
+        {
+          value: PatientModal.Encounter,
+          label: t('button.add-encounter'),
+          isDisabled: disableEncounterButton,
+        },
+        {
+          value: PatientModal.ContactTraceSearch,
+          label: t('button.add-contact-trace'),
+        }
+      );
+    }
+
+    if (InsuranceProvidersData?.length > 0) {
+      baseOptions.push({
+        value: PatientModal.Insurance,
+        label: t('button.add-insurance'),
+      });
+    }
```

Comment:
Added to support that case where we can add an insurance policy for a patient. Even though Programs Module is off.